### PR TITLE
History from 2008 - 2010.

### PR DESCRIPTION
**Pull request by WongAaronSW**
A History from 2008 - 2010.

[Preview](https://deploy-preview-89--nubook.netlify.app/team/history)

### CODE REVIEW

File: src/book/01-team/04-history/01-team-history.mdx [hunk 1]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+- Aldebaran NAOs get an upgrade (NAOv2), Dual Non-overlapping Cameras, so they can see their feet without having to bend over..
+
+**Major software changes**
+
+- Code Purge, Remove redudent AIBO Code realised by NUPlatform
+- New Git Repository, moved away from CVS
+- Introduction of Threading for NAOs, (SeeThinkThread: 30Hz, SenseMoveThread: 100Hz).
+- Blackboard based architecture
+- Soft Colour Classifications: A transition colour, that that can belong in different categories of objects, that is assigned an object with context at a later stage in the processing pipeline.
+- Advances in Line Detection, with the introduction of corner point (L and T), penalty spot, and center circle detection (ellipse fitting).
```
Comment (KipHamiltons) [inline src/book/01-team/04-history/01-team-history.mdx:71]: > Advances in Line Detection, with the introduction of corner point (L and T),

What is `(L and T)`?

File: src/book/01-team/04-history/01-team-history.mdx [hunk 2]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+## 2010: Singapore
+
+**Major hardware changes**
+
+- NAOv3: Intel ATOM 1.6GHz, 1GB Ram,
+
+**Major software changes**
+
+** Performance and Results**
+NUBots were unfortunately knocked out during second round robins, due to a loss with rUNSWift.
```
Comment (KipHamiltons) [inline src/book/01-team/04-history/01-team-history.mdx:51]: > due to a loss with rUNSWift.

Were rUNSWift on our team, or did they knock us out?

File: src/book/01-team/04-history/01-team-history.mdx (hunk 1 above)
Comment (JosephusPaye) [reply src/book/01-team/04-history/01-team-history.mdx:71]: It's the shape of the corner lines: 

image

Comment (KipHamiltons) [reply src/book/01-team/04-history/01-team-history.mdx:71]: Is that something I should have known? should we add that to the glosssary?

File: src/book/01-team/04-history/01-team-history.mdx [hunk 3]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
 | Eliminations for Quarter-Finals (Penalty Shootout) | 3      | 1      | 1   | 0    | 0    |
 | Quarter-Finals                                     | 0      | 1      | 0   | 0    | 1    |
 | Drop-Ins                                           | -5     | 8      | 4   | 1    | 3    |
+
+## 2010: Singapore
+
+**Major hardware changes**
+
+- NAOv3: Intel ATOM 1.6GHz, 1GB Ram,
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:46]:
```suggestion
- Upgraded to NAOv3: Intel ATOM 1.6GHz, 1GB Ram
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 4]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+
+**Major software changes**
+
+** Performance and Results**
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:50]:
```suggestion
** Performance and Results**

```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 5]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+| Round              | Points | Played | Won | Drew | Lost |
+| ------------------ | ------ | ------ | --- | ---- | ---- |
+| First Round Robin  | 6      | 2      | 2   | 0    | 0    |
+| Second Round Robin | 2      | 3      | 0   | 2    | 1    |
+
+## 2009: Graz, Austria
+
+**Major hardware changes**
+
+- Aldebaran NAOs get an upgrade (NAOv2), Dual Non-overlapping Cameras, so they can see their feet without having to bend over..
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:62]:
```suggestion
- Aldebaran NAOs get an upgrade (NAOv2), dual non-overlapping cameras, so they can see their feet without having to bend over.
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 6]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+
+**Major software changes**
+
+- Code Purge, Remove redudent AIBO Code realised by NUPlatform
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:66]:
```suggestion
- Code purge, removed redundant AIBO Code realised by NUPlatform
```

Comment (JosephusPaye) [reply src/book/01-team/04-history/01-team-history.mdx:66]: For clarity, do you mean that NUPlatform made some AIBO code redundant, and that code was removed?

File: src/book/01-team/04-history/01-team-history.mdx [hunk 7]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+- New Git Repository, moved away from CVS
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:67]:
```suggestion
- New Git repository, moved away from CVS
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 8]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+- Introduction of Threading for NAOs, (SeeThinkThread: 30Hz, SenseMoveThread: 100Hz).
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:68]:
```suggestion
- Introduction of threading for NAOs, (SeeThinkThread: 30Hz, SenseMoveThread: 100Hz)
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 9]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+- Blackboard based architecture
+- Soft Colour Classifications: A transition colour, that that can belong in different categories of objects, that is assigned an object with context at a later stage in the processing pipeline.
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:70]:
```suggestion
- Soft colour classifications: starting using a transition colour, that can belong in different categories of objects, that is assigned an object with context at a later stage in the processing pipeline.
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 10]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+- Advances in Line Detection, with the introduction of corner point (L and T), penalty spot, and center circle detection (ellipse fitting).
+- Improvements in Localisation as a direct result of the additional field objects seen.
+- Multiple Models Extended Kalman Filter
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:73]:
```suggestion
- Multiple model extended Kalman Filter
```

File: src/book/01-team/04-history/01-team-history.mdx (hunk 1 above)
Comment (wongosaurus) [reply src/book/01-team/04-history/01-team-history.mdx:71]: I can make it clearer when I'm improving it. "Intersecting field-lines that form a corner point with the shape of an 'L', and an Intersecting field-lines which intersects in the shape of a 'T'".

FYI, I met up former members Kenny Hong, and Steven Nicklin today, they will be able to fill in more of these Gaps, from 2004-2013. They are going to give some more information, team photos, etc. So this page will grow.

File: src/book/01-team/04-history/01-team-history.mdx (hunk 2 above)
Comment (wongosaurus) [reply src/book/01-team/04-history/01-team-history.mdx:51]: Knocked us out.

File: src/book/01-team/04-history/01-team-history.mdx (hunk 1 above)
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:71]:
```suggestion
- Advances in Line Detection, with the introduction of corner point (L and T shapes), penalty spot, and center circle detection (ellipse fitting).
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 11]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+- Goal Keeper: Right Dive, Left Dive, and Center
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:74]:
```suggestion
- Implemented Goal Keeper behaviour: right dive, left dive, and center
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 12]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+
+** Performance and Results**
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:76]:
```suggestion
** Performance and Results**

```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 13]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+| -------------- | ------ | ------ | --- | ---- | ---- |
+| Round Robins   | 13     | 5      | 4   | 1    | 0    |
+| Quarter-Finals | 0:1    | 1      | 0   | 0    | 1    |
+
+## 2008: Suzhou, China
+
+**Major hardware changes**
+
+- First year with the Aldebaran NAOs (x86 AMD GEODE 500MHz CPU, 256 MB SDRAM / 2 GB flash memory)
+- Low-Powered Walk Optimisation
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:89]:
```suggestion
- Low-powered walk optimisation
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 14]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+
+**Major Software changes**
+
+- AIBO 2007 Code base to interface with NAOQI (Aldebarans Middleware for communicating with NAOs low-level hardware).
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:93]:
```suggestion
- Updated AIBO 2007 code base to interface with NAOQI (using Aldebarans middleware for communicating with NAOs low-level hardware).
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 15]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+
+** Performance and Results**
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:95]:
```suggestion
** Performance and Results**

```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 16]
```diff
@@ -38,3 +38,66 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+NUIManoids (The University of Newcastle (NUbots) partnered with National University of Maynooth from Ireland) - First place, Standard Platform League.
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:96]: Let's highlight our win in bold 😅

```suggestion
NUIManoids (The University of Newcastle (NUbots) partnered with National University of Maynooth from Ireland) - **First place, Standard Platform League**.
```

Comment (JosephusPaye) [reply src/book/01-team/04-history/01-team-history.mdx:96]: Also, was it NUImanoids or NUmanoids? The former makes sense for University of Ireland, but I think I've seen the latter before.

**Review by JosephusPaye — CHANGES_REQUESTED** [review]
A few suggestions, mostly formatting

File: src/book/01-team/04-history/01-team-history.mdx (hunk 1 above)
Comment (JosephusPaye) [reply src/book/01-team/04-history/01-team-history.mdx:71]: That's great news!

File: src/book/01-team/04-history/01-team-history.mdx (hunk 16 above)
Comment (wongosaurus) [reply src/book/01-team/04-history/01-team-history.mdx:96]: https://www.newcastle.edu.au/__data/assets/pdf_file/0011/126002/NUbotFinalReport2008.pdf
Numanoids.

image

File: src/book/01-team/04-history/01-team-history.mdx [hunk 17]
```diff
@@ -38,3 +38,69 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+
+**Major hardware changes**
+
+- Upgraded to NAOv3: Intel ATOM 1.6GHz, 1GB Ram
+
+**Major software changes**
+
+** Performance and Results**
+
+NUBots were unfortunately knocked out during second round robins by rUNSWift.
```
Comment (ysims) [inline src/book/01-team/04-history/01-team-history.mdx:52]:
```suggestion
NUbots were unfortunately knocked out during second round robins by rUNSWift.
```

File: src/book/01-team/04-history/01-team-history.mdx [hunk 18]
```diff
@@ -38,3 +38,69 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
+
+- Updated AIBO 2007 code base to interface with NAOQI (using Aldebarans middleware for communicating with NAOs low-level hardware).
+
+** Performance and Results**
+
+NUManoids (The University of Newcastle (NUbots) partnered with National University of Maynooth from Ireland) - **First place, Standard Platform League**.
+
+| Round                            | Points | Played | Won | Drew | Lost |
+| -------------------------------- | ------ | ------ | --- | ---- | ---- |
+| Round Robins                     | 12     | 3      | 0   | 4    | 0    |
```
Comment (ysims) [inline src/book/01-team/04-history/01-team-history.mdx:103]: Played 3 but drew 4? That's impressive.

**Review by ysims — DISMISSED** [review]
LGTM

File: src/book/01-team/04-history/01-team-history.mdx [hunk 19]
```diff
@@ -66,3 +66,72 @@ NUbots made it to the quarter-finals of the competition. Overall we played 13 ga
...
 | Quarter-Finals                                     | 0      | 1      | 0   | 0    | 1    |
 | Drop-Ins                                           | -5     | 8      | 4   | 1    | 3    |
+
+## 2010: Singapore
+
+**Major hardware changes**
+
+- Upgraded to NAOv3: Intel ATOM 1.6GHz, 1GB Ram
+
+**Major software changes**
```
Comment (JosephusPaye) [inline src/book/01-team/04-history/01-team-history.mdx:76]: We can remove the heading if there's no information.

```suggestion
```

**Review by JosephusPaye — CHANGES_REQUESTED** [review]
Just one little thing

File: src/book/01-team/04-history/01-team-history.mdx (hunk 19 above)
Comment (ysims) [reply src/book/01-team/04-history/01-team-history.mdx:76]: I thought about this then decided it's probably good to have so people know we didn't forget about it, there's just nothing to put in? Although I assume there would have been _some_ change to the software 🤔

Comment (JosephusPaye) [reply src/book/01-team/04-history/01-team-history.mdx:76]: In the case of nothing to put in, we could explicitly add something like _No major software changes_. 

And yeah, there was probably some change to the software.

**Review by JosephusPaye — APPROVED** [review]
LGTM!

**Pull request merged**